src/__tests__/resources/asset.integration.spec.ts

```diff
@@ -0,0 +1,219 @@
+    externalId: 'test-child' + randomInt(),
+  };
+  let assets: AssetList;
+
+  test('create', async () => {
+    assets = await client.assets.create([childAsset, rootAsset]);
+    expect(assets[0].createdTime).toBeInstanceOf(Date);
+    expect(assets[0].lastUpdatedTime).toBeInstanceOf(Date);
+  });
+
+  test('update', async () => {
+    await client.assets.update([
+      {
+        id: assets[0].id,
+        update: {
+          name: { set: 'New name' },
+        },
+      },
+    ]);
+  });
+
+  test('delete', async () => {
+    await assets[0].delete();
+  });
+
+  test('list', async () => {
+    await client.assets
+      .list({
+        filter: {
+          name: rootAsset.name,
+          createdTime: { min: 0, max: Date.now() },
+        },
+      })
+      .autoPagingToArray({ limit: 100 });
+  });
+
+  test('search for root test asset', async () => {
+    const result = await client.assets.search({
+      search: {
+        name: rootAsset.name,
+      },
+    });
+    const asset = result[0];
+    expect(rootAsset.name).toBe(asset.name);
+    expect(rootAsset.description).toBe(asset.description);
+  });
+
+  describe('Asset Class', () => {
+    const newRootAsset = {
+      name: 'test-root',
+    };
+    const newChildAsset = {
+      name: 'test-child',
+    };
+    const newGrandChildAsset = {
+      name: 'test-grandchild',
+    };
+
+    test('create 1001 assets sequencially', async () => {
+      const childArr = new Array(1000).fill(newChildAsset);
+      const createdAssets = await client.assets.create([
+        newRootAsset,
+        ...childArr,
+      ]);
+      expect(createdAssets.length).toBe(1001);
+      await client.assets.delete(
+        createdAssets.slice(1).map(asset => ({ id: asset.id }))
+      );
+      await client.assets.delete([{ id: createdAssets[0].id }]);
+    });
+
+    test('parent', async () => {
+      const newRoot = {
+        ...newRootAsset,
+        externalId: 'test-root' + randomInt(),
+      };
+      const newChild = {
+        ...newChildAsset,
+        parentExternalId: newRoot.externalId,
+      };
+      const createdAssets = await client.assets.create([newRoot, newChild]);
+      const parent = await createdAssets[1].parent();
+      if (parent) {
+        expect(parent.id).toEqual(createdAssets[0].id);
+      }
+      const parent2 = await createdAssets[0].parent();
+      expect(parent2).toBe(null);
+      await client.assets.delete([{ id: createdAssets[0].id }]);
+    });
+
+    test('children', async () => {
+      const newRoot = {
+        ...newRootAsset,
+        externalId: 'test-root' + randomInt(),
+      };
+      const newChild = {
+        ...newChildAsset,
+        parentExternalId: newRoot.externalId,
+      };
+      const createdAssets = await client.assets.create([
+        newRoot,
+        newChild,
+        newChild,
+      ]);
+      const children = await createdAssets[0].children();
+      expect(children.length).toBe(2);
```

Comment:
This keeps failing, but it passes if i add `sleepPromise(xxxx)` inside the `.children()` method, so I guess it's just a backend this?

Reply:
We need to take a closer look at this. Retrieving children with `await sleepPromise(10000)` only gives 1, sometimes give 0 children, and that is not how it should be. What I'm I doing wrong?

Reply:
Maybe fix with this? #229

Reply:
Should I try this or leave it for now?

Reply:
If you could try to implement it with `retryInSeconds` that would be good. We don't want to have any test in the `v1` branch that sometimes fails since this will then block all future PRs...

Reply:
This is now a unit test